projects/lexical_credo/lib/lexical_credo.ex

```diff
@@ -0,0 +1,102 @@
+defmodule LexicalCredo do
+  alias Lexical.Document
+  alias Lexical.Plugin.Diagnostic
+  alias Lexical.Project
+
+  use Diagnostic, name: :lexical_credo
+  require Logger
+
+  def init do
+    with {:ok, _} <- Application.ensure_all_started(:credo) do
+      :ok
+    end
+  end
+
+  def document do
+    %Document{}
+  end
+
+  def handle(%Document{} = doc) do
+    Application.ensure_all_started(:credo)
+
+    doc_contents = Document.to_string(doc)
+
+    execution_args = [
+      "--mute-exit-status",
+      "--read-from-stdin",
+      Document.Path.absolute_from_uri(doc.uri)
+    ]
+
+    execution = Credo.Execution.build(execution_args)
+
+    with_stdin(
```

Comment:
I think we should try to keep the operation results consistent with `mix credo`, so we don't need to perform real-time credo checks for excluded files.

Reply:
The answer is that credo doesn't listen to the excludes or includes when using stdin as input.